notebooks/2022-21-03users_with_link_in_comment.ipynb

```diff
@@ -0,0 +1,203 @@
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>2990</th>\n",
+       "      <td>61630</td>\n",
+       "      <td>amitsulu</td>\n",
+       "      <td>NULL</td>\n",
+       "      <td>2016-08-30 06:49:01</td>\n",
+       "      <td>0.0</td>\n",
+       "      <td>0.0</td>\n",
+       "      <td>3.0</td>\n",
+       "      <td>0.0</td>\n",
+       "      <td>â€¢You may determine out quite a few #Packers ...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>13834</th>\n",
+       "      <td>179044</td>\n",
+       "      <td>robinsmithsters</td>\n",
+       "      <td>NULL</td>\n",
+       "      <td>2020-09-25 11:46:22</td>\n",
+       "      <td>0.0</td>\n",
+       "      <td>0.0</td>\n",
+       "      <td>3.0</td>\n",
+       "      <td>0.0</td>\n",
+       "      <td>Higher education is a lofty endeavor with as m...</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "           id         username description                date  edits  \\\n",
+       "2990    61630         amitsulu        NULL 2016-08-30 06:49:01    0.0   \n",
+       "13834  179044  robinsmithsters        NULL 2020-09-25 11:46:22    0.0   \n",
+       "\n",
+       "       reviews  comments  taxonomy  \\\n",
+       "2990       0.0       3.0       0.0   \n",
+       "13834      0.0       3.0       0.0   \n",
+       "\n",
+       "                                        comment_has_link  \n",
+       "2990   â€¢You may determine out quite a few #Packers ...  \n",
+       "13834  Higher education is a lofty endeavor with as m...  "
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "import mysql.connector as mysql\n",
+    "import re\n",
+    "import numpy as np\n",
+    "\n",
+    "db = mysql.connect(\n",
+    "    host=\"localhost\",\n",
+    "    user=\"root\",\n",
+    "    password=\"password\",\n",
+    "    port=\"3306\",\n",
+    "    database=\"serlo\"\n",
+    ")\n",
+    "\n",
+    "\n",
+    "def has_link(content):\n",
+    "    if content == None or content == \"NULL\":\n",
+    "        return False\n",
+    "    \n",
+    "    return \"http://\" in content or \"https://\" in content\n",
+    "\n",
+    "def query(sql):\n",
+    "    c = db.cursor()\n",
+    "    c.execute(sql)\n",
+    "    return c.fetchall()\n",
+    "\n",
+    "\n",
+    "def comment_has_link(user_id):\n",
+    "    user_comments = query('''select comment_vote.user_id, comment.content\n",
+    "    from comment_vote\n",
+    "    inner join comment\n",
+    "    on comment.id = comment_vote.comment_id\n",
+    "    where comment_vote.user_id = comment.author_id''' )\n",
+    "\n",
+    "    for user in user_comments:\n",
+    "        if user[0] == user_id:\n",
+    "            if has_link(user[1]):\n",
```

Comment:
In order to avoid nested if's (which may become with time difficult to read) in this case,
you could refactor this way:
`if user[0] == user_id and has_link(user[1]):`